determined-ai/determined
# fix: Selected experiments in glide table persist [WEB-1366]

## Description

After commit from Claire + discussion from Alex, this is a cleaner way for selected experiments to be stored for the new Glide Table.
Replaces #7221

The "Select All" action is a moment in time, so it (and excluded IDs list) do NOT persist.
Added a small commit, so if someone "selects all" and then a new experiment appears in the list, the "select all" checkbox will un-check to reflect not all experiments are checked.

## Test Plan

**Test on a single page of experiments (latest-main project 107) and a multi-page experiment list (uncategorized)**

Selection
- With `?f_explist_v2=on`, select alternating experiments in a list. Refresh or navigate away from the page and return; same experiments should be selected
- The count of selected experiments in the edit button, e.g. "Edit (10)" should reflect the count selected

Pagination
- On a project with many experiments, scroll down to the end to trigger infinite scroll / pagination behavior. Select an experiment very far down the list
- Refresh and you should be able to scroll down and see the same experiment selected

Select All (actual selections persisted)
- With some selected, the header above the checkbox is a [-] mark
- Click the header checkbox on the table - select all experiments and this header shows a filled check
- Reload the page; this should remember selected experiments, but the "Select All" header checkbox returns to [-] 
- Use Select All, then uncheck one row to exclude it.
- Reload the page; all should be selected except for the excluded row

New Experiments drop Select All state
- Open the experiment list and Select All; the header becomes a filled check
- Do not refresh the page! Start an experiment in the project and wait for it to appear in the list (sample CLI: `cd examples/tutorials/core_api_pytorch_mnist` ; `det -m latest-main.determined.ai:8080 e create const.yaml ./ --project_id 107` )
- When the new experiment row appears, it is not selected, and the select-all header returns to [-]

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
